### DISCUSSION TITLE
Asking what is the purpose of AllowedIssuers

### DISCUSSION THREAD

**Discussion by ronimhd** [apiman/apiman#2473] [Q&A]
Hi, I am a beginner with Apiman. 
During my POC for Apiman, I had a problem with Aoiman Vert.x gateway. After I troubleshoot, I found that there is validation regarding allowedIssuer in this code:

`if (config.getAllowedIssuers().contains(issuerEndpoint)) {
                        handler.handle(Future.failedFuture("issuer validation failed: received [" + issuerEndpoint + "]"));
                        return;
                    
}`

But I do not understand why if issuerEndpoint is member of allowedIssuer, the system throws an exception "issuer validation failed: received"
If this code is as the requirement, how to make configuration work, because the config.site is from "auth-server-url" and it become allowIssuer.

**Comment by msavy** [comment]
Can you please provide the full name of the file

**Reply by ronimhd** [reply]
That code in io.apiman.gateway.platforms.vertx3.api.auth.KeycloakDiscovery. In gateway/platforms/vertx3. Tag Branch_3.0.0.FInal

**Reply by msavy** [reply]
Thanks. 

This is code that is shadowed (coped) from the Vert.x 2.x codebase and modified slightly for our needs, so it may be that there was a glitch that was brought across and needs fixing.

I'll check later to see.

**Reply by msavy** [reply]
You are welcome to make a PR, btw!

**Reply by ronimhd** [reply]
PR is created to master
But I want to use 3.0.0.Final, how to do that?

**Reply by msavy** [reply]
You'll need to do your own build in that case.

3.1.0.Final will be out in shortly once I fix a few build issues.

**Reply by ronimhd** [reply]
I have created PR to master, can you approve it?

**Reply by msavy** [reply]
This is now merged and will go into 3.1.0.Final

**Comment by msavy** [comment]
Thanks. It needs manually testing before I can approve it. I'll respond
there when it's done.

You can still do your own build locally to use your change
